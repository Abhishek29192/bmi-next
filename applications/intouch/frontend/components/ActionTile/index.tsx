import React from "react";
import CardActionArea from "@material-ui/core/CardActionArea";
import Typography from "@bmi/typography";
import Icon from "@bmi/icon";
import { BMI } from "@bmi/logo";
import styles from "./styles.module.scss";

export type ActionTileProps = {
  title: string;
  description: string;
  // action?: ClickableAction;
};

export const ActionTile = ({ title, description }: ActionTileProps) => {
  return (
    <div className={styles.main}>
      <CardActionArea className={styles.clickable}>
        <Icon source={BMI} className={styles.logo} />
        <div className={styles.text}>
          <Typography variant="h5" className={styles.title}>
            {title}
          </Typography>
          <Typography variant="subtitle2">{description}</Typography>
        </div>
      </CardActionArea>
    </div>
  );
};
