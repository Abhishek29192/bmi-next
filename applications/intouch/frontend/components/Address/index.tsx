import React from "react";
import { Address as AddressType } from "@bmi/intouch-api-types";
import { gql } from "@apollo/client";

export type AddressProps = {
  address: Pick<
    AddressType,
    "firstLine" | "secondLine" | "town" | "region" | "country" | "postcode"
  >;
};

export const Address = ({ address }: AddressProps) => (
  <div data-testid="address">
    {["firstLine", "secondLine", "postcode", "town", "region", "country"]
      .filter((addressLine) => !!address[addressLine])
      .map((addressLine) => (
        <div key={addressLine} data-testid={`addressLine-${addressLine}`}>
          {address[addressLine]}
        </div>
      ))}
  </div>
);

export const AddressLinesFragment = gql`
  fragment AddressLinesFragment on Address {
    firstLine
    secondLine
    town
    region
    country
    postcode
  }
`;
