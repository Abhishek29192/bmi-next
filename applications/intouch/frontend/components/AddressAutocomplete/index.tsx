import React, { useEffect, useState } from "react";
import GoogleAutocomplete, {
  Props as GoogleAutocompleteProps
} from "@bmi/google-autocomplete";
import GoogleApi, { loadGoogleApi, GeocoderResult } from "@bmi/google-api";
import GoogleMap, { Props as GoogleMapProps } from "@bmi/google-map";
import { Address, Point } from "@bmi/intouch-api-types";
import { addressFromPlaceApiResponse } from "./placeApiResponseToAddress";
import styles from "./styles.module.scss";

type AddressAutocompleteProps = GoogleAutocompleteProps & {
  onAddressSelected: (address: Partial<Address>) => any;
  mapProps: GoogleMapProps;
  locationBias?: Point;
};

// One challenges of location biasing is to find an appropriate radius for different markets
// In order to customize this bias radius, we could potentially in future add a field to the market record, which specifies the radius
const DEFAULT_LOCATION_BIAS_RADIUS = 100 * 1000; // in metres, currently the same for all markets

export const AddressAutocomplete = ({
  onAddressSelected,
  googleAutocompleteOptions = {},
  mapProps,
  locationBias,
  ...rest
}: AddressAutocompleteProps) => {
  const [googleApi, setGoogleApi] = useState(null);

  const initialise = async () => {
    await loadGoogleApi(process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY, [
      "places"
    ]);
    setGoogleApi(typeof window?.google !== "undefined" ? window.google : null);
  };

  const handlePlaceChange = (placeApiResponse: GeocoderResult) => {
    if (googleApi) {
      onAddressSelected(addressFromPlaceApiResponse(placeApiResponse));
    }
  };

  useEffect(() => {
    initialise();
  }, []);

  return (
    <GoogleApi.Provider value={googleApi}>
      <GoogleAutocomplete
        // Some results seem to not be returned by Places API (compared to same Google Maps).
        // It seems like the fuzzy search on Places API is not optimal (it's much better with Google Maps)
        onPlaceChange={handlePlaceChange}
        getOptionSelected={(place) => place.description}
        googleAutocompleteOptions={{
          ...googleAutocompleteOptions,
          location:
            googleApi && locationBias
              ? new googleApi.maps.LatLng(locationBias.x, locationBias.y)
              : undefined,
          // radius is just for biasing purposes: locations outside of the center + radius
          // will be returned anyway, but with lower ranking
          radius:
            googleApi && locationBias ? DEFAULT_LOCATION_BIAS_RADIUS : undefined
        }}
        startAdornmentIcon="LocationOn"
        {...rest}
      />
      <div className={styles.mapContainer}>
        {googleApi?.maps && <GoogleMap {...mapProps} />}
      </div>
    </GoogleApi.Provider>
  );
};
