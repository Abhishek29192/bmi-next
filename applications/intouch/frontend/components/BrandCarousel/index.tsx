import React from "react";
import styles from "./styles.module.scss";

export type BrandCarouselProps = {
  children?: React.ReactNode | React.ReactNode[];
};

export const BrandCarousel = ({ children }: BrandCarouselProps) => {
  return <div className={styles.main}>{children}</div>;
};
