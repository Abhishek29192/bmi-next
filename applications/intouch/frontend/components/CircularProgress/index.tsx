import MuiCircularProgress, {
  CircularProgressProps as MuiCircularProgressProps
} from "@material-ui/core/CircularProgress";
import React from "react";
import styles from "./styles.module.scss";

export type CircularProgressProps = {
  backgroundProps?: MuiCircularProgressProps;
  railProps?: MuiCircularProgressProps;
  value: MuiCircularProgressProps["value"];
};

export const CircularProgress = ({
  backgroundProps,
  railProps,
  value
}: CircularProgressProps) => {
  return (
    <div
      className={styles.circularProgressContainer}
      data-testid="circular-progress"
    >
      <MuiCircularProgress
        size={120}
        thickness={6}
        variant="determinate"
        className={styles.circularProgress}
        {...backgroundProps}
      />
      <MuiCircularProgress
        className={styles.circularProgressRail}
        size={114}
        thickness={4}
        variant="determinate"
        {...railProps}
        value={100}
      />
      <div className={styles.circularProgressLabel}>{`${value}%`}</div>
    </div>
  );
};
