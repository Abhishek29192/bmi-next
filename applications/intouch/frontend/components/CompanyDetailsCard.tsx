import React, { useState } from "react";
import { useTranslation } from "next-i18next";
import EditIcon from "@material-ui/icons/Edit";
import { gql } from "@apollo/client";
import Button from "@bmi/button";
import Dialog from "@bmi/dialog";
import Form from "@bmi/form";
import TextField from "@bmi/text-field";
import CompanyDetails from "@bmi/company-details";
import { useUpdateCompanyDetailsMutation } from "graphql/generated/hooks";
import AccessControl from "../lib/permissions/AccessControl";
import { GetCompanyQuery } from "../graphql/generated/operations";

type CompanyType = GetCompanyQuery["company"];

const getCompanyData = (
  company: GetCompanyQuery["company"]
): { name: string; aboutUs?: string; details: any } => {
  if (!company) {
    return {
      name: "Company not found",
      details: []
    };
  }

  const details = [
    {
      type: "cta",
      text: "Get directions",
      action: { model: "htmlLink" as "htmlLink", href: company.website },
      label: "Get directions"
    },
    {
      type: "phone",
      text: company.phone,
      action: {
        model: "htmlLink" as "htmlLink",
        href: `tel:${company.phone}`
      },
      label: "Telephone"
    },
    {
      type: "email",
      text: company.publicEmail,

      action: {
        model: "htmlLink" as "htmlLink",
        href: `mailto:${company.publicEmail}`
      },

      label: "Email"
    },
    {
      type: "website",
      text: "Visit website",
      action: {
        model: "htmlLink" as "htmlLink",
        href: company.website
      },
      label: "Website"
    },
    {
      type: "content",
      label: "Type of roof",
      text: <b>Flat</b>
    },
    {
      type: "roofProLevel",
      label: "BMI RoofPro Level",
      level: "expert"
    }
  ];

  return {
    name: company.name,
    aboutUs: company.aboutUs,
    details
  };
};

export const CompanyDetailsFragment = gql`
  fragment CompanyDetailsFragment on Company {
    id
    name
    phone
    website
    aboutUs
    publicEmail
    phone
    website
    companyMembers {
      nodes {
        id
      }
    }
  }
`;

export const UPDATE_COMPANY = gql`
  mutation updateCompanyDetails($input: UpdateCompanyInput!) {
    updateCompany(input: $input) {
      company {
        ...CompanyDetailsFragment
      }
    }
  }
`;

const EditDialog = ({
  company,
  isOpen,
  onCloseClick
}: {
  company: CompanyType;
  isOpen: boolean;
  onCloseClick: () => void;
}) => {
  const [updateCompany] = useUpdateCompanyDetailsMutation();

  const handleSave = (event, values) => {
    event.preventDefault();
    updateCompany({
      variables: {
        input: {
          patch: values,
          id: company.id
        }
      }
    });
    onCloseClick && onCloseClick();
  };

  return (
    <Dialog open={isOpen} onCloseClick={onCloseClick}>
      <Dialog.Title hasUnderline>Edit Company Details</Dialog.Title>
      <Dialog.Content>
        <Form onSubmit={handleSave}>
          <Form.Row>
            <TextField
              name="name"
              isRequired
              variant="outlined"
              label="Company Name"
              fullWidth
            />
          </Form.Row>
          <Form.ButtonWrapper>
            <Form.SubmitButton>Save & close</Form.SubmitButton>
          </Form.ButtonWrapper>
        </Form>
      </Dialog.Content>
      <Dialog.Actions confirmLabel={"Save & close"} />
    </Dialog>
  );
};

type Props = {
  company: CompanyType;
};

const CompanyDetailsCard = ({ company }: Props) => {
  const { t } = useTranslation("common");
  const [isEditDialogOpen, setEditDialogOpen] = useState(false);
  const { name, aboutUs, details } = getCompanyData(company);

  const companyMemberIds = company.companyMembers.nodes.map(({ id }) => id);

  return (
    <CompanyDetails name={name} details={details}>
      {aboutUs || ""}
      <AccessControl
        dataModel="company"
        action="edit"
        extraData={{ companyMemberIds }}
      >
        <Button
          variant="text"
          color="primary"
          onClick={() => setEditDialogOpen(true)}
          startIcon={<EditIcon />}
        >
          {t("Edit")}
        </Button>
      </AccessControl>
      <EditDialog
        company={company}
        isOpen={isEditDialogOpen}
        onCloseClick={() => setEditDialogOpen(false)}
      />
    </CompanyDetails>
  );
};

export default CompanyDetailsCard;
