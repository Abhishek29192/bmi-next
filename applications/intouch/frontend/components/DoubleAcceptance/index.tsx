import Confirmation from "./Confirmation";
import FormContainer from "./FormContainer";

export { Confirmation, FormContainer };
