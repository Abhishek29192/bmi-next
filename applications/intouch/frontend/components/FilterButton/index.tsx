import React from "react";
import { Chip } from "@bmi/components";
import styles from "./styles.module.scss";

export type FilterButtonProps = {
  label: string;
  isActive?: boolean;
  onClick?: () => void;
};

export const FilterButton = ({
  label,
  isActive,
  onClick
}: FilterButtonProps) => {
  return (
    <div className={styles.filterButton}>
      <Chip
        type="selectable"
        isSelected={isActive}
        onClick={() => onClick && onClick()}
      >
        {label}
      </Chip>
    </div>
  );
};
