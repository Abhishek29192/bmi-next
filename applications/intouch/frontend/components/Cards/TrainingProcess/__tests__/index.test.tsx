import { ThemeProvider } from "@bmi/components";
import { render } from "@testing-library/react";
import React from "react";
import { TrainingProcessCard } from "..";

describe("TrainingProcessCard component", () => {
  it("renders correctly", () => {
    const data = {
      items: [
        {
          pageSubHeading: "The Training Process",
          step1Heading: "Step 1",
          step1SubHeading: "Find the right training",
          step1Description: `Search our extensive library of classroom training, workshops,
      online webinars and e-learning courses to find what you need and
      take your skills to the next level.`,
          step2Heading: "Step 2",
          step2SubHeading: "Increase your skills",
          step2Description: `Continue personal development to hone your skills in areas where you
      need additional knowledge.`,
          step3Heading: "Step 3",
          step3SubHeading: "Get certified",
          step3Description: `Develop your training achievements by collecting BMI certificates,
      which you can additionally share with others. They will always be
      visible on your profile.`
        }
      ]
    };

    const { container } = render(
      <ThemeProvider>
        <TrainingProcessCard data={data} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
