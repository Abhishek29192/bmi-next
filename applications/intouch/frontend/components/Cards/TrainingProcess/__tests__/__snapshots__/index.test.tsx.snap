// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingProcessCard component renders correctly 1`] = `
<div
  class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  style="margin-top: 1.5rem;"
>
  <div
    class="MuiCardContent-root"
  >
    <h4
      class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
    >
      The Training Process
    </h4>
    <div
      class="body"
    >
      <div
        class="subcontent"
      >
        <div
          class="heading"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <h5
            class="MuiTypography-root Typography MuiTypography-h5"
          >
            Step 1
          </h5>
        </div>
        <div
          style="margin-bottom: 1.5rem;"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Find the right training
          </h6>
          <p
            class="MuiTypography-root Typography MuiTypography-body1"
          >
            Search our extensive library of classroom training, workshops,
      online webinars and e-learning courses to find what you need and
      take your skills to the next level.
          </p>
        </div>
      </div>
      <div
        class="subcontent"
      >
        <div
          class="heading"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
            />
          </svg>
          <h5
            class="MuiTypography-root Typography MuiTypography-h5"
          >
            Step 2
          </h5>
        </div>
        <div
          style="margin-bottom: 1.5rem;"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Increase your skills
          </h6>
          <p
            class="MuiTypography-root Typography MuiTypography-body1"
          >
            Continue personal development to hone your skills in areas where you
      need additional knowledge.
          </p>
        </div>
      </div>
      <div
        class="subcontent"
      >
        <div
          class="heading"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
            />
          </svg>
          <h5
            class="MuiTypography-root Typography MuiTypography-h5"
          >
            Step 3
          </h5>
        </div>
        <div
          style="margin-bottom: 1.5rem;"
        >
          <h6
            class="MuiTypography-root Typography MuiTypography-h6"
          >
            Get certified
          </h6>
          <p
            class="MuiTypography-root Typography MuiTypography-body1"
          >
            Develop your training achievements by collecting BMI certificates,
      which you can additionally share with others. They will always be
      visible on your profile.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
